pkg/resources/etcd/statefulset.go

```diff
@@ -95,28 +100,42 @@ func StatefulSetCreator(data etcdStatefulSetCreatorData, enableDataCorruptionChe
 			}
 			set.Spec.Template.Spec.ServiceAccountName = rbac.EtcdLauncherServiceAccountName
 
-			set.Spec.Template.Spec.InitContainers = []corev1.Container{
-				{
-					Name:            "etcd-launcher-init",
-					Image:           data.EtcdLauncherImage() + ":" + resources.KUBERMATICCOMMIT,
-					ImagePullPolicy: corev1.PullIfNotPresent,
-					Command:         []string{"/bin/cp", "/etcd-launcher", "/opt/bin/"},
-					VolumeMounts: []corev1.VolumeMount{
-						{
-							Name:      "launcher",
-							MountPath: "/opt/bin/",
+			// luancher feature gate was added later in the process. So there is the possiblity to have some clusters with
```

Comment:
s/luancher/launcher/

Reply:
I'm not totally convinced we need this logic. The point is that if we want to fallback, disabling the `etcd-launcher` feature is not enough, and we also need to manually remove the init container from the manifest.

It's true that we have already RC versions with this feature activated by default, but they are not intended for production. I think we are still in time to avoid this.

Reply:
If we don't absolutely need this to fix an existing etcd cluster, I agree with @irozzo-1A and we should skip adding this logic. We would probably never get rid of it, even long after any possible migration is done.